### ISSUE TITLE
Returning non namespace scoped state?

### ISSUE THREAD

**Issue by MikeRock** [Sawtaytoes/Ghadyani-Framework-Redux-Utils#1]
Hi. I've been looking through the code for the createNamespaceReducerCreator and found this bit [#76](https://github.com/Sawtaytoes/Ghadyani-Framework-Redux-Utils/blob/ce752f72300cb209e844227c204dcb1bff8fe47e/utils/createNamespaceReducerCreator.js#L76)

It implies that the namespaces reducer returns the base non namespace reducer state.
Am I seeing this correctly and is it the intended behavior ?

Thanks.

**Comment by Sawtaytoes** [comment]
I see what you're saying. It makes sense that it should be returning `prevNamespaceReducer` instead of `prevState`.

All unit tests are passing even with this bug in place, and I found out it's because I wasn't properly testing for it. In fact, it never reached that line of code in any unit test because `prevState` wasn't primed. That line is also a performance improvement to avoid deep equals. Without it, the code should still work.

Either way, **I went ahead and fixed the issue** and added 2 more unit tests, one of which covers that scenario.

Not only that, I refactored that section to make it easier to read and also changed the following logic around the `initialState` since there's a major performance improvement to be gained by not doing a deep equals check of `nextState` and `initialState`.

At this point, I'm wondering if that early-return is even necessary anymore as the removal of the deep equals check makes it pretty useless.

I really appreciate you bringing this to my attention and encourage you to point out any other issues you find in the future!
